src/PIL/IcoImagePlugin.py

```diff
@@ -138,55 +136,70 @@ def __init__(self, buf):
 
         # Get headers for each item
         for i in range(self.nb_items):
-            s = buf.read(16)
-
-            icon_header = {
-                "width": s[0],
-                "height": s[1],
-                "nb_color": s[2],  # No. of colors in image (0 if >=8bpp)
-                "reserved": s[3],
-                "planes": i16(s, 4),
-                "bpp": i16(s, 6),
-                "size": i32(s, 8),
-                "offset": i32(s, 12),
-            }
-
-            # See Wikipedia
-            for j in ("width", "height"):
-                if not icon_header[j]:
-                    icon_header[j] = 256
-
-            # See Wikipedia notes about color depth.
-            # We need this just to differ images with equal sizes
-            icon_header["color_depth"] = (
-                icon_header["bpp"]
-                or (
-                    icon_header["nb_color"] != 0
-                    and ceil(log(icon_header["nb_color"], 2))
-                )
-                or 256
-            )
+            self.entry.append(self._read_icon_header(buf))
 
-            icon_header["dim"] = (icon_header["width"], icon_header["height"])
-            icon_header["square"] = icon_header["width"] * icon_header["height"]
+        if not self.entry:
+            msg = "No images were found"
```

Comment:
```suggestion
            msg = "No icons were found"
```
Rather than the more generic word "images", how about something specific to this format, to hopefully avoid confusion with our Image class? This would also then be similar to "No cursors were found"

Reply:
As mentioned in my earlier comment, the future followup to this will base CUR image support on this plugin, so this moves the error to the common "images" (which covers both "icons" and "cursors"). If you like, I can make this a class field, so the deriving cursor plugin can change the message to `No cursors were found`..?